source/common/stats/symbol_table_impl.h

```diff
@@ -115,6 +118,9 @@ class SymbolTableImpl : public SymbolTable {
   // TODO(ambuc): There might be an optimization here relating to storing ranges of freed symbols
   // using an Envoy::IntervalSet.
   std::stack<Symbol> pool_;
+
+  // This must be called during both encode() and free().
+  mutable Thread::MutexBasicLockable lock_;
```

Comment:
It looks to me like the access to this class is also protected by a mutex; maybe you don't need both?

Reply:
The symbol table must have a local mutex, since it has to be able to protect itself during `.toString()` operations. So, good catch -- the mutex at the callsite in `HeapStatDataAllocator` only needs protects the local set of `HeapStatData*` s (typedef'd as `StatSet`). I'll make this change.